R/caretStack.R

```diff
@@ -165,16 +178,24 @@ predict.caretStack <- function(
     excluded_class_id <- 0L
   }
 
+  # Make sure all original_features are present in newdata
+  if (!is.null(object$original_features)) {
+    check_original_features(object$original_features, newdata, training = FALSE)
+  }
+
   # Get predictions from the submodels on the new data
   # We need theres if there's newdata, for passing the base model predictions to the stack model
   # We also need these if we're calculting standard errors for the predictions
-  sub_model_preds <- if (!is.null(newdata) || se) {
-    stats::predict(
+  if (!is.null(newdata) || se) {
+    sub_model_preds <- stats::predict(
       object$models,
       newdata = newdata,
       verbose = verbose,
       excluded_class_id = object[["excluded_class_id"]]
     )
+    if (!is.null(object$original_features)) {
+      sub_model_preds <- cbind(sub_model_preds, newdata[, object$original_features])
```

Comment:
I think it makes sense to call `check_original_features` again here too

Reply:
I previously called check_original_features in lines 182-184, but it would be clearer to move the call to just above this line.

Reply:
yup— I see that now.  I just mis-read this code block.